File: p_success_mc.py
from scipy.stats import norm
import numpy as np
import random

def p_success_mc(n, m, alpha, xi, Nsim = 10000):

    #This function returns the ordinal optimisation probability of success for
    #the Gaussian case, using Monte-Carlo simulation

    count = 0

    X_thres = norm.ppf(alpha)

    for j in range(Nsim):
        X = norm.rvs(size=n)
        Y = xi*norm.rvs(size=n)
        Z = X + Y
    
        #Count number of successes
        ind = np.argsort(Z)
        if (min(X[ind[0:m]]) <= X_thres):
            count += 1
    
    p = count/Nsim
    #Return the standard error as a second argument
    se = np.sqrt(p*(1 - p)/Nsim)
    
    return p, se

File: p_success_mc.m
function pout = p_success_mc(n, m, alpha, xi, Nsim)

%This function returns the ordinal optimisation probability of success for
%the Gaussian case, using Monte-Carlo simulation

if (nargin < 5)
    Nsim = 20000; %number of simulations
end

%unconditional
count = 0;
for i = 1:Nsim
    
    X = randn(n, 1);
    Y = xi*randn(n, 1);
    
    Z = X + Y;
    
    [~, indsort] = sort(Z, 'ascend');
    
    if (any(X(indsort(1:m)) <= norminv(alpha)))
        count = count + 1;
    end
    
end

pout = count/Nsim;

end

